src/types/ant.ts

```diff
@@ -0,0 +1,186 @@
+      },
+ */
+export const ArweaveTxIdSchema = z
+  .string({
+    description: 'Arweave Transaction ID',
+  })
+  .refine((val) => ARWEAVE_TX_REGEX.test(val), {
+    message: 'Must be an Arweave Transaction ID',
+  });
+
+export const IntegerStringSchema = z
+  .string({
+    description: 'Integer String',
+  })
+  .refine(
+    (val) => {
+      const num = parseInt(val);
+      return Number.isInteger(num) && num >= 0;
+    },
+    { message: 'Must be a non negative integer string' },
+  );
+export const AntRecordSchema = z.object({
+  transactionId: ArweaveTxIdSchema.describe('The Target ID of the undername'),
+  ttlSeconds: z.number(),
+});
+export type AoANTRecord = z.infer<typeof AntRecordSchema>;
+
+export const AntRecordsSchema = z.record(z.string(), AntRecordSchema);
+export const AntControllersSchema = z.array(
+  ArweaveTxIdSchema.describe('Controller address'),
+);
+export const AntBalancesSchema = z.record(
+  ArweaveTxIdSchema.describe('Holder address'),
+  z.number(),
+);
+
+export const AntStateSchema = z.object({
+  Name: z.string().describe('The name of the ANT.'),
+  Ticker: z.string().describe('The ticker symbol for the ANT.'),
+  Denomination: z
+    .number()
+    .describe(
+      'The number of decimal places to use for the ANT. Defaults to 0 if not set representing whole numbers.',
+    )
+    .min(0, { message: 'Denomination must be a non-negative number' }),
+  Owner: ArweaveTxIdSchema.describe('The Owners address.'),
+  Controllers: AntControllersSchema.describe(
+    'Controllers of the ANT who have administrative privileges.',
+  ),
+  Records: AntRecordsSchema.describe('Records associated with the ANT.'),
+  Balances: AntBalancesSchema.describe(
+    'Balance details for each address holding the ANT.',
+  ),
+  Logo: ArweaveTxIdSchema.describe('Transaction ID of the ANT logo.'),
+  TotalSupply: z
+    .number()
+    .describe('Total supply of the ANT in circulation.')
+    .min(0, { message: 'Total supply must be a non-negative number' }),
+  Initialized: z
+    .boolean()
+    .describe('Flag indicating whether the ANT has been initialized.'),
+  ['Source-Code-TX-ID']: ArweaveTxIdSchema.describe(
+    'Transaction ID of the Source Code for the ANT.',
+  ),
+});
+
+export type AoANTState = z.infer<typeof AntStateSchema>;
+
+export const AntInfoSchema = z.object({
+  Name: z.string().describe('The name of the ANT.'),
+  Owner: ArweaveTxIdSchema.describe('The Owners address.'),
+  ['Source-Code-TX-ID']: ArweaveTxIdSchema.describe(
+    'Transaction ID of the Source Code for the ANT.',
+  ),
+  Ticker: z.string().describe('The ticker symbol for the ANT.'),
+  ['Total-Supply']: IntegerStringSchema.describe(
+    'Total supply of the ANT in circulation.',
+  ),
+  Logo: ArweaveTxIdSchema.describe('Transaction ID of the ANT logo.'),
+  Denomination: IntegerStringSchema.describe(
+    'The number of decimal places to use for the ANT. Defaults to 0 if not set representing whole numbers.',
+  ),
+  HandlerNames: z
```

Comment:
let's modify this in the ants to be `Handlers` to match the ar-io-network-process

Reply:
_I found 2 ants with different names for the handlers. It is in fact currently "Handlers". We updated it couple weeks ago and now we have old ants with "HandlerNames" and new ones with "Handlers". I currently set them both to optional and marked the HandlerNames one as deprecated in the description for the schema field. Zod doesnt have a built in way to show deprecations there and dont think we want to blow up here.
This is making me wonder if maybe we should have a flag in the constructor for "strict" apis that do blow up on bad data. Or maybe something like ant.strict.getState()_

Reply:
I found 2 ants with different names for the handlers. It is in fact currently "Handlers". We updated it couple weeks ago and now we have old ants with "HandlerNames" and new ones with "Handlers". I currently set them both to optional and marked the HandlerNames one as deprecated in the description for the schema field. Zod doesnt have a built in way to show deprecations there and dont think we want to blow up here.
This is making me wonder if maybe we should have a flag in the constructor for "strict" apis that do blow up on bad data. Or maybe something like ant.strict.getState()